engine/src/main/scala/cromwell/engine/workflow/lifecycle/WorkflowExecutionActor.scala

```diff
@@ -67,12 +68,15 @@ object WorkflowExecutionActor {
 
     /** Checks if the workflow is completed by scanning through the executionStore */
     def isWorkflowComplete: Boolean = {
-      def isDone(executionStatus: ExecutionStatus): Boolean = executionStatus == ExecutionStatus.Done
+      def isDone(executionStatus: ExecutionStatus): Boolean = (executionStatus == ExecutionStatus.Done) || (executionStatus == ExecutionStatus.Preempted)
       executionStore.values.forall(isDone)
     }
 
     /** Updates the status of a a job by the new entry */
     def updateExecutionStoreStatus(entry: (JobKey, ExecutionStatus)): WorkflowExecutionActorData = this.copy(executionStore = executionStore + entry)
+
+    def addNewJob(jobKey: JobKey, executionStatus: ExecutionStatus = ExecutionStatus.NotStarted): WorkflowExecutionActorData =
+      this.copy(executionStore = executionStore + (jobKey -> executionStatus))
```

Comment:
Ahh yes you're right. I spent a lot of time to not create the very function (the one that was there before this commit) but gave up and had to create one. Then when I changed the stuff to keep the Preempted jobs,  I just changed it little. I can definitely remove this now. Thanks!